File: SDKs/amazonaws.com/elasticache/2015-02-02/python/src/sdk/models/operations/get_rebootcachecluster.py
from dataclasses import dataclass, field
from datetime import date, datetime
from marshmallow import fields
import dateutil.parser
from typing import List,Optional
from enum import Enum

class GetRebootCacheClusterActionEnum(str, Enum):
    REBOOT_CACHE_CLUSTER = "RebootCacheCluster"

class GetRebootCacheClusterVersionEnum(str, Enum):
    TWO_THOUSAND_AND_FIFTEEN_02_02 = "2015-02-02"


@dataclass
class GetRebootCacheClusterQueryParams:
    action: GetRebootCacheClusterActionEnum = field(metadata={'query_param': { 'field_name': 'Action', 'style': 'form', 'explode': True }})
    cache_cluster_id: str = field(metadata={'query_param': { 'field_name': 'CacheClusterId', 'style': 'form', 'explode': True }})
    cache_node_ids_to_reboot: List[str] = field(metadata={'query_param': { 'field_name': 'CacheNodeIdsToReboot', 'style': 'form', 'explode': True }})
    version: GetRebootCacheClusterVersionEnum = field(metadata={'query_param': { 'field_name': 'Version', 'style': 'form', 'explode': True }})
    

@dataclass
class GetRebootCacheClusterHeaders:
    x_amz_algorithm: Optional[str] = field(default=None, metadata={'header': { 'field_name': 'X-Amz-Algorithm', 'style': 'simple', 'explode': False }})
    x_amz_content_sha256: Optional[str] = field(default=None, metadata={'header': { 'field_name': 'X-Amz-Content-Sha256', 'style': 'simple', 'explode': False }})
    x_amz_credential: Optional[str] = field(default=None, metadata={'header': { 'field_name': 'X-Amz-Credential', 'style': 'simple', 'explode': False }})
    x_amz_date: Optional[str] = field(default=None, metadata={'header': { 'field_name': 'X-Amz-Date', 'style': 'simple', 'explode': False }})
    x_amz_security_token: Optional[str] = field(default=None, metadata={'header': { 'field_name': 'X-Amz-Security-Token', 'style': 'simple', 'explode': False }})
    x_amz_signature: Optional[str] = field(default=None, metadata={'header': { 'field_name': 'X-Amz-Signature', 'style': 'simple', 'explode': False }})
    x_amz_signed_headers: Optional[str] = field(default=None, metadata={'header': { 'field_name': 'X-Amz-SignedHeaders', 'style': 'simple', 'explode': False }})
    

@dataclass
class GetRebootCacheClusterRequest:
    headers: GetRebootCacheClusterHeaders = field()
    query_params: GetRebootCacheClusterQueryParams = field()
    

@dataclass
class GetRebootCacheClusterResponse:
    content_type: str = field()
    status_code: int = field()
    body: Optional[bytes] = field(default=None)
    

File: SDKs/amazonaws.com/elasticache/2015-02-02/typescript/src/sdk/models/operations/getrebootcachecluster.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";


export enum GetRebootCacheClusterActionEnum {
    RebootCacheCluster = "RebootCacheCluster"
}

export enum GetRebootCacheClusterVersionEnum {
    TwoThousandAndFifteen0202 = "2015-02-02"
}


export class GetRebootCacheClusterQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=Action" })
  action: GetRebootCacheClusterActionEnum;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=CacheClusterId" })
  cacheClusterId: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=CacheNodeIdsToReboot" })
  cacheNodeIdsToReboot: string[];

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=Version" })
  version: GetRebootCacheClusterVersionEnum;
}


export class GetRebootCacheClusterHeaders extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Algorithm" })
  xAmzAlgorithm?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Content-Sha256" })
  xAmzContentSha256?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Credential" })
  xAmzCredential?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Date" })
  xAmzDate?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Security-Token" })
  xAmzSecurityToken?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Signature" })
  xAmzSignature?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-SignedHeaders" })
  xAmzSignedHeaders?: string;
}


export class GetRebootCacheClusterRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  queryParams: GetRebootCacheClusterQueryParams;

  @SpeakeasyMetadata()
  headers: GetRebootCacheClusterHeaders;
}


export class GetRebootCacheClusterResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  body?: Uint8Array;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;
}
